src/unix/loop.c

```diff
@@ -36,6 +37,15 @@ int uv_loop_init(uv_loop_t* loop) {
   memset(loop, 0, sizeof(*loop));
   loop->data = saved_data;
 
+  mloop = (uv__metrics_loop_t*) uv__calloc(1, sizeof(uv__metrics_loop_t));
+  if (mloop == NULL)
+    return UV_ENOMEM;
+
```

Comment:
Maybe allocating `uv__metrics_loop_t`  only if/when `UV_LOOP_IDLE_TIME` is set?

Reply:
From @bnoordhuis comment in https://github.com/libuv/libuv/issues/2506#issuecomment-540050665:

> I think that can be addressed by malloc-ing memory.

Since that's basically what I'm doing with `uv__metrics_loop_t`, maybe it's not a bad time to create a `struct` that will hold all future would-be ABI changes. Something like `uv__loop_internal_t`?

@bnoordhuis @saghul Thoughts?